Templates/templates/Java/requests_extensions/BaseMethodRequest.java.tt

```diff
@@ -128,10 +117,11 @@ import <#=mainNamespace#>.<#=c.GetPackagePrefix()#>.<#=c.AsOdcmMethod().TypePara
     /**
      * Gets the <#=c.ReturnType()#>
      *
-     * @param callback the callback to be called after success or failure
+     * @return a future with the result
      */
-    public void get(@Nonnull final ICallback<? super <#=c.ReturnType()#>> callback) {
-        send(HttpMethod.GET, callback, null);
+    @Nonnull
```

Comment:
I think we should remove `@Nonnull` as there could be method responses that contain an empty object. If a method returns an empty object, or a null object. The default is nullable=true for return types. How does Java client handle a `{}` for response versus an empty response? 


<Action Name="invite" IsBound="true" xmlns="http://docs.oasis-open.org/odata/ns/edm">   <Parameter Name="bindingParameter" Type="Collection(graph.participant)" />   <Parameter Name="participants" Type="Collection(graph.invitationParticipantInfo)" Nullable="false" />   <Parameter Name="clientContext" Type="Edm.String" Unicode="false" />   <ReturnType Type="graph.inviteParticipantsOperation" /> </Action>
--
<Action Name="renewGroup" IsBound="true" xmlns="http://docs.oasis-open.org/odata/ns/edm">   <Parameter Name="bindingParameter" Type="Collection(graph.groupLifecyclePolicy)" Nullable="false" />   <Parameter Name="groupId" Type="Edm.String" Nullable="false" Unicode="false" />   <ReturnType Type="Edm.Boolean" Nullable="false" /> </Action>

Reply:
Whatever the service responds, we'll still have a future being returned. (Or an exception thrown) the result of that future might be null but that's not what this method is returning directly

Reply:
Got it.